### PR TITLE
sanity test simplifications

### PR DESCRIPTION
**Pull request by kripken**
After the recent browser test improvements, the sanity tests are now the slowest. This makes them 6 minutes faster (10%).

 * Remove testing for falling back to pre-fastcomp. We don't support that.
 * Remove `test_embuilder` - @juj made the point that it was redundant, as we do test building them all anyhow (during the build step), so it just mostly just checks that the file output names are as expected, and that the `ALL` target is complete, which doesn't seem worth it.

### CODE REVIEW

File: tests/test_sanity.py [hunk 1]
```diff
@@ -329,8 +326,6 @@ def test_llvm_fastcomp(self):
 
     restore_and_set_up()
 
-    self.check_working([EMCC] + MINIMAL_HELLO_WORLD + ['-s', 'ASM_JS=0'], '''Compiler settings are incompatible with fastcomp. You can fall back to the older compiler core, although that is not recommended''')
-
```
Comment (sbc100) [inline tests/test_sanity.py]: If you are going to remove this test, why not also remove the error message in emcc.py that cause it.  Better still, revert this line and make a separate CL for removing that.

Comment (kripken) [reply tests/test_sanity.py]: Ok.
